esp-hal/src/interrupt/riscv.rs

```diff
@@ -667,6 +683,156 @@ mod classic {
+    }
+
+    /// Enable a CPU interrupt
+    ///
+    /// # Safety
+    ///
+    /// Make sure there is an interrupt handler registered.
+    pub unsafe fn enable_cpu_interrupt(which: CpuInterrupt) {
+        let cpu_interrupt_number = which as usize;
+        let intr_cntrl = intr_cntrl(crate::get_core(), cpu_interrupt_number);
+
+        unsafe {
+            let mut val = InterruptControl(intr_cntrl.read_volatile());
+            val.set_enabled(true);
+            intr_cntrl.write_volatile(val.0);
+        }
+    }
+
+    /// Set the interrupt kind (i.e. level or edge) of an CPU interrupt
+    ///
+    /// This is safe to call when the `vectored` feature is enabled. The
+    /// vectored interrupt handler will take care of clearing edge interrupt
+    /// bits.
+    pub fn set_kind(core: Cpu, which: CpuInterrupt, kind: InterruptKind) {
+        let cpu_interrupt_number = which as usize;
+        unsafe {
+            let intr_cntrl = intr_cntrl(core, cpu_interrupt_number);
+            let mut val = InterruptControl(intr_cntrl.read_volatile());
+            val.set_trigger(match kind {
+                InterruptKind::Level => 0b00,
+                InterruptKind::Edge => 0b10,
+            });
+            intr_cntrl.write_volatile(val.0);
+        }
+    }
+
+    /// Set the priority level of an CPU interrupt
+    ///
+    /// # Safety
+    ///
+    /// Great care must be taken when using the `vectored` feature (enabled by
+    /// default). Avoid changing the priority of interrupts 1 - 15 when
+    /// interrupt vectoring is enabled.
+    pub unsafe fn set_priority(core: Cpu, which: CpuInterrupt, priority: Priority) {
+        let cpu_interrupt_number = which as usize;
+        let intr_cntrl = intr_cntrl(core, cpu_interrupt_number);
+
+        unsafe {
+            let mut val = InterruptControl(intr_cntrl.read_volatile());
+            val.set_priority(priority as u8);
+            intr_cntrl.write_volatile(val.0);
+        }
+    }
+
+    /// Clear a CPU interrupt
+    #[inline]
+    pub fn clear(core: Cpu, which: CpuInterrupt) {
+        let cpu_interrupt_number = which as usize;
+
+        unsafe {
+            let intr_cntrl = intr_cntrl(core, cpu_interrupt_number);
+            let mut val = InterruptControl(intr_cntrl.read_volatile());
+            val.set_pending(false);
+            intr_cntrl.write_volatile(val.0);
+        }
+    }
+
+    /// Get interrupt priority
+    #[inline]
+    pub(super) fn get_priority_by_core(core: Cpu, cpu_interrupt: CpuInterrupt) -> Priority {
+        let cpu_interrupt_number = cpu_interrupt as usize;
+
+        unsafe {
+            let intr_cntrl = intr_cntrl(core, cpu_interrupt_number);
+            let val = InterruptControl(intr_cntrl.read_volatile());
+
+            core::mem::transmute(val.priority())
+        }
+    }
+
+    /// Get interrupt priority - called by assembly code
+    #[inline]
+    pub(super) unsafe extern "C" fn get_priority(cpu_interrupt: CpuInterrupt) -> Priority {
```

Comment:
Without implementing these the rest of the interrupt code is somewhat useless, can we do this in a follow up PR and with a full impl?